src/explore-education-statistics-admin/src/pages/release/data/__tests__/ReleaseApiDataSetLocationsMappingPage.test.tsx

```diff
@@ -418,10 +442,623 @@ describe('ReleaseApiDataSetLocationsMappingPage', () => {
+
+      await waitFor(() =>
+        expect(screen.getByText('Unmapped locations')).toBeInTheDocument(),
+      );
+
+      expect(
+        screen.getByRole('link', {
+          name: 'There is 1 unmapped local authority',
+        }),
+      ).toBeInTheDocument();
+
+      // unmapped table
+      const unmappedTable = screen.getByRole('table', {
+        name: 'Local Authorities 1 unmapped location 2 mapped locations',
+      });
+      const unmappedLocation = within(unmappedTable).getAllByRole('row')[1];
+      const unmappedLocationCells =
+        within(unmappedLocation).getAllByRole('cell');
+      expect(within(unmappedLocationCells[0]).getByText('Location 2'));
+      expect(within(unmappedLocationCells[1]).getByText('Unmapped'));
+      expect(within(unmappedLocationCells[3]).getByText('N/A'));
+
+      // new locations table
+      expect(
+        screen.getByRole('heading', {
+          name: 'New locations (4) No action required',
+        }),
+      ).toBeInTheDocument();
+      expect(
+        screen.getByRole('heading', {
+          name: 'Local Authorities (2) Show',
+        }),
+      ).toBeInTheDocument();
+      const newLocationsTable = screen.getByTestId('new-localAuthority');
+      const newLocationRows = within(newLocationsTable).getAllByRole('row');
+      expect(newLocationRows).toHaveLength(3);
+      const newLocationCells = within(newLocationRows[1]).getAllByRole('cell');
+      expect(
+        within(newLocationCells[0]).getByText(
+          'Not applicable to current data set',
+        ),
+      );
+      expect(within(newLocationCells[1]).getByText('Location 6'));
+
+      await user.click(
+        within(unmappedLocation).getByRole('button', {
+          name: 'Edit mapping for Location 2',
+        }),
+      );
+
+      expect(
+        await screen.findByText('Map existing location'),
+      ).toBeInTheDocument();
+
+      const modal = within(screen.getByRole('dialog'));
+
+      await user.click(modal.getByLabelText('Location 6'));
+
+      expect(
+        apiDataSetVersionService.updateLocationsMapping,
+      ).not.toHaveBeenCalled();
+
+      await user.click(
+        modal.getByRole('button', { name: 'Update location mapping' }),
+      );
+
+      await waitFor(() =>
+        expect(
+          screen.queryByText('Map existing location'),
+        ).not.toBeInTheDocument(),
+      );
+
+      jest.runAllTimers();
+
+      await waitFor(() => {
+        expect(
+          apiDataSetVersionService.updateLocationsMapping,
+        ).toHaveBeenCalledWith('draft-version-id', {
+          updates: [
+            {
+              candidateKey: 'Location6Key',
+              level: 'localAuthority',
+              sourceKey: 'Location2Key',
+              type: 'ManualMapped',
+            },
+          ],
+        });
+      });
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText('There is 1 unmapped local authority'),
+        ).not.toBeInTheDocument();
+      });
```

Comment:
The first one, `expect(screen.queryByText('Map existing location'),).not.toBeInTheDocument()`, has to be done before `jest.runAllTimers();` otherwise the tests fail consistently when all tests are run (it's fine if just this suite is run), which seems to be some oddness related to fakeTimers but I couldn't figure out what.
 
I've combined the other two into one `waitFor`.